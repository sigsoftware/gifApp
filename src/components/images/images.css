.gridImages {
    display: grid;
    grid-template-columns: repeat(auto-fill, 230px);
    gap: 40px;
    padding: 35px;
    justify-content: center;
}

@media (max-width:560px) {
    .gridImages {
        grid-template-columns: 100%,
    }
}

h2{
    text-align: center;
    color: aliceblue;
    font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
}