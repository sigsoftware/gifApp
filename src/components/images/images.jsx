import React from 'react';
import "./images.css";
import "animate.css";
import PropTypes from "prop-types";

const Images = (props) => {
    return (
        <>
            <h2>Resultados para <b>{String(props?.term).toLocaleUpperCase()}</b></h2>
            <ul className='gridImages animate__animated animate__bounceInUp'>{props.children}</ul>
        </>
    )
}

Images.defaultProps = {
    term: "",
}

Images.propTypes = {
    term: PropTypes.string.isRequired,
}

export { Images };