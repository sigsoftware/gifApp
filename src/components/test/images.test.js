
import { shallow } from "enzyme";
import { Images } from "./../images/images";

describe("Pruebas a realizar al componente de Images", () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Images term={"Test"} />)
    })

    test("Testear que la pagina se renderiza correctamente", async () => {
        expect(wrapper).toMatchSnapshot();
    })

    test("Testear que el elemento tiene en el className animate__animated animate__bounceInUp", async () => {
        expect(String(wrapper.find('ul').prop('className')).split(" ")[2]).toBe("animate__bounceInUp");
    })

    test("Testear el valor del elemento H2", async () => {
        let valueH2 = wrapper.find('h2').text().trim();
        expect(String(valueH2).split(" ")[2]).toBe("TEST");
    })

    test("Testear si esta en el elemento className la cadena String de animate_animated",()=>{
        let ul = wrapper.find('ul');
        let className = ul.prop("className");
        expect(className.includes("animate__animated")).toBe(true);
    })

    test("Testear que no esta en el elemento className la cadena String de animate_animated",()=>{
        let ul = wrapper.find('ul');
        let className = ul.prop("className");
        expect(className.includes("animate_faraona")).not.toBe(true);
    })
});