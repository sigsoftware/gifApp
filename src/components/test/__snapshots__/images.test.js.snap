// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a realizar al componente de Images Testear que la pagina se renderiza correctamente 1`] = `
<Fragment>
  <h2>
    Resultados para 
    <b>
      TEST
    </b>
  </h2>
  <ul
    className="gridImages animate__animated animate__bounceInUp"
  />
</Fragment>
`;
